### PR TITLE
Add pool option to cron jobs

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

**Review by Cisphyx — COMMENTED** [review]
Seems pretty straightforward, do we want to add an option for pool=True when adding a cron job as well?

**Review by MichaelSquires — APPROVED** [review]

**Comment by MichaelSquires** [comment]
> Seems pretty straightforward, do we want to add an option for pool=True when adding a cron job as well?

And show the pool flag when running `cron.list`?

File: synapse/lib/agenda.py [hunk 1]
```diff
@@ -815,19 +821,33 @@ async def _runJob(self, user, appt):
...
-                    count += 1
+
+                async for mesg in self.core.storm(appt.query, opts=opts):
+
+                    if mesg[0] == 'node':
+                        count += 1
+
+                    elif mesg[0] == 'err':
+                        excname, errinfo = mesg[1]
+                        # TOOD should these even come out of the storm runtime anymore?
```
Comment (vEpiphyte) [inline synapse/lib/agenda.py:841]: I'd argue "yes" since I like having more debug data when things break ;)

File: synapse/lib/agenda.py [hunk 2]
```diff
@@ -366,6 +369,7 @@ def unpack(cls, stor, val):
         appt = cls(stor, val['iden'], val['recur'], val['indx'], val['query'], val['creator'], recs, nexttime=val['nexttime'], view=val.get('view'))
         appt.doc = val.get('doc', '')
         appt.name = val.get('name', '')
+        appt.pool = val.get('pool', False)
```
Comment (vEpiphyte) [inline synapse/lib/agenda.py:372]: we can set `pool=val.get('pool', False)` in the `cls(...)` call?

**Review by vEpiphyte — CHANGES_REQUESTED** [review]
Can we add the `pool` status to the `cron.stat` command output?

File: synapse/lib/agenda.py (hunk 2 above)
Comment (invisig0th) [reply synapse/lib/agenda.py:372]: This whole thing should be converted to an info dict, but that's a separate PR. I went with the existing convention for this one.

**Comment by invisig0th** [comment]
> > Seems pretty straightforward, do we want to add an option for pool=True when adding a cron job as well?
> 
> And show the pool flag when running `cron.list`?

Adding to `cron.stat` the `cron.list` output already seems pretty cluttered. Think it's worth it?

**Comment by vEpiphyte** [comment]
I think adding it to to cron.list would be more clutter than needed?

**Review by Cisphyx — DISMISSED** [review]

**Review by mikemoritz — APPROVED** [review]

File: synapse/lib/agenda.py (hunk 1 above)
Comment (vEpiphyte) [reply synapse/lib/agenda.py]: remove comment?

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**